### PR TITLE
[CU-f92zkb] Add tournament details - match table

### PR DESCRIPTION
**Pull request by Zummek**
TODO: mobile view, waiting for api service/currentUser store

165573884_358377278744409_1138820908703231487_n
164878535_279018170335451_871995279026282141_n

### CODE REVIEW

**Comment by Zummek** [comment]
Task linked: [CU-f92zkb Create tournament details view - games table tab](https://app.clickup.com/t/f92zkb)

**Comment by eastemployeer** [comment]
Inconsistency spotted :) Mobile view card should have margins to main container (as it is done on desktop view)

**Comment by Zummek** [comment]
@eastemployeer 
> Inconsistency spotted :) Mobile view card should have margins to main container (as it is done on desktop view)

I disabled margin on mobile to save space(more space for data), hmm i set margin to `sm` and looks like this:
image

**Comment by Zummek** [comment]
Still waiting for CR @eastemployeer  @Buglik

File: shared/types/Tournament/index.d.ts [hunk 1]
```diff
@@ -26,7 +26,7 @@ export interface Match {
 export default interface Tournament {
   id?: string;
   name: string;
-  ownerMSId: string;
+  owner: User;
```
Comment (eastemployeer) [inline shared/types/Tournament/index.d.ts]: Comparing to our main branch and every CRUD functions that we currently implement, it should stay like it is on main branch.

File: web-app/src/classes/Match.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  // date: Date;
+
+  constructor(data: IMatch) {
+    this.id = data.id;
+    this.sideA = data.sideA;
+    this.sideB = data.sideB;
+    this.isFinished = data.isFinished;
+    // this.date = data.date;
+  }
+}
```
Comment (eastemployeer) [inline web-app/src/classes/Match.ts]: And this class is for? Can't we just make object that's type is Match (from sharedTypes?) and don't create this extra class which makes everything just less understandable?

File: web-app/src/components/tournament/details/OutcomeTableItem.vue [hunk 3]
```diff
@@ -0,0 +1,216 @@
+<template>
```
Comment (eastemployeer) [inline web-app/src/components/tournament/details/OutcomeTableItem.vue]: Name of the file is inappropriate. Should be ResultTableItem. Not Outcome.

File: shared/types/Tournament/index.d.ts (hunk 1 above)
Comment (Zummek) [reply shared/types/Tournament/index.d.ts]: I prepared this branch to work with resolving userId by backend. I think this change should happen in a few days

File: web-app/src/classes/Match.ts (hunk 2 above)
Comment (Zummek) [reply web-app/src/classes/Match.ts]: This is trash, useless. I forgot to delete this

File: web-app/src/components/tournament/details/OutcomeTableItem.vue (hunk 3 above)
Comment (Zummek) [reply web-app/src/components/tournament/details/OutcomeTableItem.vue]: Why result? `Outcome table item` points to the item of table of outcomes. `Result` sounds like result of some function. We used 'outcomes' naming
hmm may we should change name to outcome => outcomes

Comment (eastemployeer) [reply web-app/src/components/tournament/details/OutcomeTableItem.vue]: No. Word outcome is wrong. Google --> outcome meaning

Comment (Zummek) [reply web-app/src/components/tournament/details/OutcomeTableItem.vue]: Wynik, rezultat. This word was used by company in doc

Comment (Zummek) [reply web-app/src/components/tournament/details/OutcomeTableItem.vue]: Imo "score" its the best

**Review by eastemployeer — APPROVED** [review]

**Pull request merged**